december.js

```diff
@@ -2716,459 +2703,724 @@ function checkEffects() {
+                'https://cdn.discordapp.com/attachments/483446980605902848/781295276778323988/1363841058659.gif'
+            ]
+        },
+    }
+    ///////////////////////////////////////////
+    // "Public" Static methods
+    ///////////////////////////////////////////
+    static init() {
+        PresentsEffect.state = {
+            is_on: false,
+            enabled: true,
+            timeout: null,
+            level: PresentsEffect.levels[0],
+            version: PresentsEffect.versions['normal'],
+        }
+        PresentsEffect.container = document.createElement('div');
+        document.documentElement.appendChild(PresentsEffect.container);
+    }
+
+    static disable() {
+        PresentsEffect.state.enabled = false;
+    }
+
+    static enable() {
+        PresentsEffect.state.enabled = true;
+    }
+
+    static stop() {
+        PresentsEffect.state.is_on = false;
+    }
+
+    static addElement(element) {
+        PresentsEffect.container.appendChild(element);
+    }
+
+    static handleCommand(message_parts, otherArgs) {
+
+        // No parse, right now just one version
+        // TODO: add different H levels and loli options
+
+        // Disable presents after the timeout. If there is already one, reset the timer
+        if (PresentsEffect.state.timeout) {
+            clearTimeout(PresentsEffect.state.timeout);
+        }
+        PresentsEffect.state.timeout =
+            setTimeout(PresentsEffect.stop, PresentsEffect.presents_duration_s * 1000);
+
+        // Only start the padoru animation if it is not already started
+        if (PresentsEffect.state.is_on) {
+            return;
+        }
+        PresentsEffect.state.is_on = true;
+        PresentsEffect._face_animation();
+        PresentsEffect._run_presents_animation();
+    }
+    ///////////////////////////////////////////
+    // Timed Static methods
+    ///////////////////////////////////////////
+    static _face_animation(){
+        if (!PresentsEffect.state.is_on) {
+            return;
+        }
+        const face_img = PresentsEffect.state.version.padoru;
+
+        const face_effect = document.createElement('img')
+        face_effect.classList.add('c-effect__presents-face');
+        face_effect.src = face_img;
+        PresentsEffect.addElement(face_effect);
+
+        const fn = () =>{
+            face_effect.parentElement.removeChild(face_effect);
+            face_effect.removeEventListener('animationend', fn);
+        }
+        face_effect.addEventListener('animationend', fn);
+    }
+
+    static _run_presents_animation() {
+        const create_fn = (is_left) => {
+            if (!PresentsEffect.state.is_on) {
+                return;
+            }
+
+            PresentsEffect._create_present(is_left);
+            setTimeout(create_fn.bind(null,!is_left), PresentsEffect.state.level.spawn_rate);
+        };
+        setTimeout(create_fn.bind(null, true), PresentsEffect.state.level.spawn_rate);
+    }
```

Comment:
Using `.bind` is fine but using `() => {}` is preferred

```suggestion
    static _run_presents_animation() {
        const create_fn = (is_left) => {
            if (!PresentsEffect.state.is_on) {
                return;
            }

            PresentsEffect._create_present(is_left);
            setTimeout(() => create_fn(!is_left), PresentsEffect.state.level.spawn_rate);
        };
        setTimeout(() => create_fn(true), PresentsEffect.state.level.spawn_rate);
    }
```